MetaMask/metamask-extension
# refactor sentryHooks object

## Explanation



This PR reinstates methods in relation to the extension state, that are used when debugging e2e tests

- Rename the `sentryHooks` object `stateHooks`
- Fix `metamaskGetState` method used to get fixture data
- Fix `getCleanAppState` method used to add the state to the ci artifacts for failed tests

## Screenshots/Screencaps



### Before



### After



metamaskGetState

<img width="1594" alt="metamaskGetState" src="https://user-images.githubusercontent.com/53189696/200906708-d6c0cedb-603d-43d5-9aba-6b88f2925002.png">

download state logs
<img width="1312" alt="state logs" src="https://user-images.githubusercontent.com/53189696/200907256-dabdccf5-3ec8-4b61-bdcd-17833d2eab34.png">

getCleanAppState - ci test-failure-state.json
<img width="1594" alt="test failure state" src="https://user-images.githubusercontent.com/53189696/200907982-bf6b7b56-b600-41d0-95d5-fbce0d4b3f62.png">

getSentryState
<img width="1594" alt="sentry appState" src="https://user-images.githubusercontent.com/53189696/200913192-9b012704-cd0b-4cc0-a9a4-5668514730f5.png">


## Manual Testing Steps



scenario 1: check state logs download and state inside
- login to the extension
- click the account menu icon
- click settings
- click advanced
- click download state logs
- check the downloaded file contains state

scenario 2: check state dump for e2e testing
- build a test build of the extension
- login to the extension
- open the background console
- enter `copy(await stateHooks.metamaskGetState()) `
- check the state is copied to the clipboard

scenario 3: check failed tests state
- run a single e2e test, change an assertion so the test fails
- look in the directory ./test-artifacts/${browser}/${test}/test-failure-state.json
- check the file is populated with state

scenario 4: check sentry state
- run `yarn test:e2e:single test/e2e/tests/errors.spec.js --browser chrome`
- check the test passes as it has assertions on the state sent to sentry

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
